### PR TITLE
Include the block variations on the inserter selector

### PR DESCRIPTION
**Pull request by youknowriad**
The way block variations are included in the inserter is very hacky. It also results in a number of issues:

 - Variations need to be included in all situations manually
 - The search algorithm doesn't treat them as their own unit causing bad results
 - We had to add hacky code to make the inserter "limit" work.

This PR solves this by just moving the inserter variations to the inserter items selector.

### CODE REVIEW

File: packages/block-editor/src/store/selectors.js [hunk 1]
```diff
@@ -1365,6 +1365,29 @@ const canIncludeBlockTypeInInserter = ( state, blockType, rootClientId ) => {
...
+	title: variation.title || item.title,
+	description: variation.description || item.description,
+	// If `example` is explicitly undefined for the variation, the preview will not be shown.
+	example: variation.hasOwnProperty( 'example' )
+		? variation.example
+		: item.example,
+	initialAttributes: {
+		...item.initialAttributes,
+		...variation.attributes,
+	},
```
Comment (ntsekouras) [inline packages/block-editor/src/store/selectors.js:1387]:
```suggestion
	},
	keywords: variation.keywords,
```

**Review by ntsekouras — APPROVED** [review]
Great work here Riad! 💯. I've left a comment about adding `keywords` in `getItemFromVariation` to include those as well. You can test it by searching for example `music`.

File: packages/block-editor/src/store/selectors.js [hunk 2]
```diff
@@ -1502,7 +1526,30 @@ export const getInserterItems = createSelector(
...
-		return [ ...blockTypeInserterItems, ...reusableBlockInserterItems ];
+		// Exclude any block type item that is to be replaced by a default
+		// variation.
+		const visibleBlockTypeInserterItems = blockTypeInserterItems.filter(
+			( { variations = [] } ) =>
+				! variations.some( ( { isDefault } ) => isDefault )
+		);
+
+		// Fill `variationsToAdd` until there are as many items in total as
+		// `limit`.
```
Comment (mcsf) [inline packages/block-editor/src/store/selectors.js:1537]:
```suggestion
```

**Review by mcsf — COMMENTED** [review]
Happy to see this refactor.

**Review by mcsf — APPROVED** [review]

**Comment by mcsf** [comment]
> Great work here Riad! 💯. I've left a comment about adding `keywords` in `getItemFromVariation` to include those as well. You can test it by searching for example `music`.

Good catch. :)

**Pull request merged**